test/integration/report_test.rb

```diff
@@ -938,13 +938,6 @@ def setup
     assert_error_resp_with_exc(ThreeScale::Backend::ProviderKeyOrServiceTokenRequired.new)
   end
 
-  test 'returns 400 when param key encoding is not valid utf-8' do
-    post '/transactions.xml', "\xf0\x90\x28\xbc" => 1
-
-    assert_equal 400, last_response.status
-    assert_equal ThreeScale::Backend::NotValidData.new.to_xml, last_response.body
-  end
```

Comment:
After https://github.com/rack/rack/pull/1797, this doesn't happen anymore.

Reply:
I don't quite understand how that PR fixes this behavior... but if this is now considered a valid input, maybe let's keep the test, and just change the expected response?